Services/Contact/classes/class.ilContactAppEventListener.php

```diff
@@ -37,7 +37,19 @@ public static function handleEvent(string $a_component, string $a_event, array $
             $user->setId((int) $a_parameter['usr_id']);
 
             $mailingLists = new ilMailingLists($user);
+            foreach ($mailingLists->getAll() as $mailingList) {
+                $mailingList->delete();
+            }
+            $assigned = $mailingLists->getAssigned();
             $mailingLists->deleteAssignments();
+            foreach ($assigned as $ass) {
+                if (ilBuddyList::getInstanceByUserId((int) $a_parameter['usr_id'])->getLinkedRelations()->isEmpty()) {
+                    $related = new ilMailingLists(new ilObjUser($ass->getUserId()));
```

Comment:
Creating user instances is very expensive, we should avoid this in scenarios where dozens of user accounts can potentially be deleted at once.

Please use simple SQL statements within `ilMailingLists`.